main.js

```diff
@@ -178,4 +178,319 @@ const data = [
+    for (let i = 0; i < daysInMonth + dayOfWeek + x; i++) {
+        if (i % daysInWeek == 0) {
+            month.push([])
+        }
+        let last = month.length - 1
+        if (i < dayOfWeek) {
+            month[last].push(daysInMonth - dayOfWeek + i + 1)
+        } else if (i >= daysInMonth + dayOfWeek) {
+            month[last].push(i % daysInMonth - dayOfWeek + 1)
+        } else {
+            month[last].push(i - dayOfWeek + 1);
+        }
+    }
+    return month;
+}
+console.log(getCalendarMonth(30, 7, 4));
+
+/*lesson 7*/
+const obj1 = {
+    a: 'a',
+    b: {
+        a: 'a',
+        b: 'b',
+        c: {
+            a: 1,
+        },
+    },
+};
+const obj2 = {
+    b: {
+        c: {
+            a: 1,
+        },
+        b: 'b',
+        a: 'a',
+    },
+    a: 'a',
+};
+const obj3 = {
+    a: {
+        c: {
+            a: 'a',
+        },
+        b: 'b',
+        a: 'a',
+    },
+    b: 'b',
+};
+const deepEqual = (object1, object2) => {
+    let object1_names = Object.getOwnPropertyNames(object1); //obj1_names - Array of children of obj1
+    let object2_names = Object.getOwnPropertyNames(object2);
+    if (object1_names.length != object2_names.length)
+        return false;
+    for (let i = 0; i < object1_names.length; i++) {
+        if (!(object1_names[i] in object2))
+            return false;
+        if (!(object2_names[i] in object1))
+            return false;
+        let object1_value = object1[object1_names[i]];
+        let object2_value = object2[object1_names[i]];
+        if ((typeof object1_value) != (typeof object2_value))
+            return false;
+        if (object1_value instanceof Object) {
+            if (!deepEqual(object1_value, object2_value))
+                return false;
+        } else if (object1_value !== object2_value) return false;
+        object1_value = object1[object2_names[i]];
+        object2_value = object2[object2_names[i]];
+        if ((typeof object1_value) != (typeof object2_value))
+            return false;
+        if (object1_value instanceof Object) {
+            if (!deepEqual(object1_value, object2_value))
+                return false;
+        } else if (object1_value !== object2_value)
+            return false;
+    }
+    return true;
+}
+console.log(deepEqual(obj1,obj2));
+console.log(deepEqual(obj1,obj3));
+
+function getCalendarMonth(daysInMonth, daysInWeek, dayOfWeek, checkInDate, checkOutDate) {
+    if (dayOfWeek >= daysInWeek)
+        throw Error(`${dayOfWeek} >= ${daysInWeek}`)
+    if (checkInDate > daysInMonth || checkOutDate > daysInMonth)
+        throw Error('Check in range is invalid')
+    if (checkOutDate < checkInDate)
+        throw Error(`${checkOutDate} < ${checkInDate}`)
+
+    let month = []
+    let x = (daysInWeek - ((daysInMonth + dayOfWeek) % daysInWeek)) % daysInWeek // count of days to complete last week.
+    for (let i = 0; i < daysInMonth + dayOfWeek + x; i++) {
+        if (i % daysInWeek == 0) {
```

Comment:
why are you using '==' instead of '==='? it can result to unexpected type coercion